packages/ember-glimmer/tests/integration/components/dynamic-components-test.js

```diff
@@ -292,4 +308,335 @@ moduleFor('Components test: dynamic components', class extends RenderingTest {
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('foo-bar Caracas arepas!');
+
+    this.runTask(() => set(this.context, 'location', 'Loisaida'));
+
+    this.assertText('foo-bar-baz Loisaida arepas!');
+
+    this.runTask(() => set(this.context, 'location', 'Caracas'));
+
+    this.assertText('foo-bar Caracas arepas!');
+  }
+
+  ['@htmlbars component helper with actions'](assert) {
+    this.registerComponent('inner-component', {
+      template: 'inner-component {{yield}}',
+      ComponentClass: Component.extend({
+        classNames: 'inner-component',
+        didInsertElement() {
+          // trigger action on click in absence of app's EventDispatcher
+          this.$().on('click', () => {
+            this.sendAction('somethingClicked');
+          });
+        },
+        willDestroyElement() {
+          this.$().off('click');
+        }
+      })
+    });
+
+    let actionTriggered = 0;
+    this.registerComponent('outer-component', {
+      template: '{{#component componentName somethingClicked="mappedAction"}}arepas!{{/component}}',
+      ComponentClass: Component.extend({
+        classNames: 'outer-component',
+        componentName: 'inner-component',
+        actions: {
+          mappedAction() {
+            actionTriggered++;
+          }
+        }
+      })
+    });
+
+    this.render('{{outer-component}}');
+
+    assert.equal(actionTriggered, 0, 'action was not triggered');
+
+    this.runTask(() => {
+      this.$('.inner-component').trigger('click');
+    });
+
+    assert.equal(actionTriggered, 1, 'action was triggered');
+  }
+
+  ['@test nested component helpers'](assert) {
+    this.registerComponent('foo-bar', { template: 'yippie! {{attrs.location}} {{yield}}' });
+    this.registerComponent('baz-qux', { template: 'yummy {{attrs.location}} {{yield}}' });
+    this.registerComponent('corge-grault', { template: 'delicious {{attrs.location}} {{yield}}' });
+
+    this.render('{{#component componentName1 location=location}}{{#component componentName2 location=location}}arepas!{{/component}}{{/component}}', {
+      componentName1: 'foo-bar',
+      componentName2: 'baz-qux',
+      location: 'Caracas'
+    });
+
+    this.assertText('yippie! Caracas yummy Caracas arepas!');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('yippie! Caracas yummy Caracas arepas!');
+
+    this.runTask(() => set(this.context, 'location', 'Loisaida'));
+
+    this.assertText('yippie! Loisaida yummy Loisaida arepas!');
+
+    this.runTask(() => set(this.context, 'componentName1', 'corge-grault'));
+
+    this.assertText('delicious Loisaida yummy Loisaida arepas!');
+
+    this.runTask(() => {
+      set(this.context, 'componentName1', 'foo-bar');
+      set(this.context, 'location', 'Caracas');
+    });
+
+    this.assertText('yippie! Caracas yummy Caracas arepas!');
+  }
+
+  ['@htmlbars component with dynamic name argument resolving to non-existent component'](assert) {
```

Comment:
fails with `@glimmer`: `	
TypeError: Cannot read property 'manager' of undefined`